gitlab-runner/src/uploader.rs

```diff
@@ -50,7 +54,7 @@ fn zip_thread(mut rx: mpsc::Receiver<UploadRequest>) {
                     let r = zip.finish();
                     drop(zip);
                     let reply = match r {
-                        Ok(_) => Ok(buf),
+                        Ok(_) => temp.rewind().map(|()| temp),
```

Comment:
It's incredibly unlikely that rewind would fail; but still would be good to add a lot message for that case

Reply:
Won't this already be propagating the error back to the other end of the channel? Or did I just misunderstand the logic flow here :sweat_smile: (If we're going to log here, it might be better to do it for the general case where `Err(_) = reply` for consistency)

Reply:
RIght i guess we could map this into an Err to be returned and report it at the other end of the channel. That might be nicer indeed (if you can't rewind a file, is there any hope the receiver can do anything with it.. and if it can, would we expect it to do the right thing)..

I guess my main point; lets avoid blindly ignoring the error ;)

Reply:
Nevermind you're using map to replace the Ok member of the result to do what i was asking for; my bad